dictdiffer/__init__.py

```diff
@@ -124,7 +124,13 @@ def diff(first, second, node=None, ignore=None, path_limit=None, expand=False,
             for value in ignore
         }
 
-    node = node or []
+    if ignore:
```

Comment:
can we merge the code with the previous condition `if isinstance(ignore, list):`?

Something around these lines:
```
if ignore:
    def _process_ignore_value(value):
        # TODO ...

    ignore = type(ignore)(_process_ignore_value(value) for value in ignore)
```

Reply:
oki done :)